boards/FSM/Src/App/App_FlowMeter.c

```diff
@@ -20,11 +31,21 @@ struct FlowMeter *App_FlowMeter_Create(float (*get_flow_rate)(void))
 
     struct FlowMeter *const flow_meter = &flow_meters[alloc_index++];
     flow_meter->get_flow_rate          = get_flow_rate;
+    flow_meter->set_periodic_signal    = set_periodic_signal;
+    flow_meter->can_tx                 = can_tx;
 
     return flow_meter;
 }
 
-float App_FlowMeter_ReadFlowRate(struct FlowMeter *flow_meter)
+void App_FlowMeter_Tick(
```

Comment:
`Tick` should only tick one flow meter, not two

Reply:
I split the function into `App_FlowMeter_TickPrimary` and `App_FlowMeter_TickSecondary`. I did this b/c I couldn't think of a simpler way of calling the `SetPeriodicSignal` pertaining to each flow meter w/o coupling with CAN. Please lmk what you think.

Reply:
That is not how we should do it. The app flow meter code should not know anything about primary and secondary flow meters. You should be exposing a getter for the flow rate and we set the periodic CAN signal in the state machine. An example can be found in imd PR.

Reply:
fixed